src/Categories/Adjoint/Parametric.agda

```diff
@@ -0,0 +1,150 @@
+module Categories.Adjoint.Parametric where
+
+open import Level
+
+open import Data.Product
+open import Categories.Category.Core using (Category)
+open import Categories.Functor using (Functor; _∘F_) renaming (id to idF)
+open Categories.Functor.Functor using (F₀; F₁; homomorphism; identity; F-resp-≈)
+open import Categories.Adjoint
+open import Categories.Category.Construction.Functors
+open import Categories.Diagram.Cowedge
+open import Categories.NaturalTransformation.Dinatural
+open import Categories.Functor.Bifunctor
+open import Categories.Morphism.Properties
+open import Categories.Functor.Bifunctor.Properties
+open import Categories.NaturalTransformation
+open import Categories.NaturalTransformation.NaturalIsomorphism
+open import Categories.Morphism.Reasoning as MorphismReasoning
+import Categories.Morphism.Reasoning as MR
+open Categories.NaturalTransformation.NaturalTransformation using (η)
+open import Function.Equality using (_⟨$⟩_; _⟶_; cong; _∘_)
+
+open import Categories.Adjoint.Equivalents
+
+private
+  variable
+    o ℓ e : Level
+    C D E : Category o ℓ e
+
+-- A parametric adjunction is a family of functors Lc ⊣ Rc indexed by an object c of
+-- a category C; equivalently, a pair of functors L : C x D → E and R : Cᵒ x E → D
+-- such that for every c in C Lc = L(c,-) ⊣ R(c,-) = Rc.
+record ParametricAdjoint {C D E : Category o ℓ e} (L : Functor C (Functors D E)) (R : Functor (Category.op C) (Functors E D)) : Set (o ⊔ ℓ ⊔ e) where
+  private
+    module C = Category C
+    module D = Category D
+    module E = Category E
+    open MR E
+    module L = Functor L
+    module R = Functor R
+    module R₀ A = Functor (R.F₀ A)
```

Comment:
As you end up using components of `R₀` and `L₀`, the results end up being not-very-readable.

One suggestion: name the arguments to `ParametricAdjoint` LL and RR, that would free up L and R to be used here. [Feel free to do something different to improve readability, which is my real concern.]

Reply:
I agree, `L.₀.₁` is not really readable. I'd opt to use double (2) letters whenever we refer to a functor in two (2) parameters; we still need to use `L` and `R` in the usual way more often than the two-parameters case, in any case.